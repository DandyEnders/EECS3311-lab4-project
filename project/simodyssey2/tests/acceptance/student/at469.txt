land
wormhole
play
move(W)
land
move(E)
move(E)
move(SE)
move(SE)
play
move(N)
wormhole
move(SE)
move(N)
move(S)
move(NE)
move(S)
move(W)
move(SE)
move(W)
move(S)
wormhole
move(E)
test(11,20,20,38,50)
pass
move(E)
move(NE)
move(NW)
move(NW)
wormhole
wormhole
move(N)
abort
move(W)
move(NW)
move(NW)
wormhole
move(N)
wormhole
move(NW)
move(N)
liftoff
liftoff
pass
abort
move(SE)
move(SE)
move(NW)
move(SW)
abort
move(E)
