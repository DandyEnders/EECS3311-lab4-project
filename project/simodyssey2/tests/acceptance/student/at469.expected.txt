  state:0.0, ok
  Welcome! Try test(3,5,7,15,30)
->land
  state:0.1, error
  Negative on that request:no mission in progress.
->wormhole
  state:0.2, error
  Negative on that request:no mission in progress.
->play
  state:1.0, mode:play, ok
  Movement:none
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    E---   *---   P---   ----   PP--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    ----   ----   W---   ----   P---   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    MY--   ----   O---   PW--   ----   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   M*--   P---   P*--   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   ----   *---   ----   PW--   
->move(W)
  state:2.0, mode:play, ok
  Movement:
    [0,E]:[1,1,1]->[1,5,3]
    [3,P]:[1,5,2]->[5,4,1]
    [8,P]:[4,3,1]->[3,2,1]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   *---   P---   ----   P-E-   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    ----   ----   W---   ----   P---   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    MY--   P---   O---   PW--   ----   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   M*--   ----   P*--   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   ----   *---   P---   PW--   
->land
  state:2.1, mode:play, error
  Negative on that request:no yellow dwarf at Sector:1:5
->move(E)
  state:3.0, mode:play, ok
  Movement:
    [0,E]:[1,5,3]->[1,1,1]
    [5,M]:[3,1,1]->[3,5,1]
    [8,P]:[3,2,1]->[2,1,1]
    [10,B]:[5,1,1]->[5,2,1]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    E---   *---   P---   ----   P---   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    P---   ----   W---   ----   P---   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   ----   O---   PW--   M---   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   M*--   ----   P*--   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    -W--   B---   *---   P---   PW--   
->move(E)
  state:4.0, mode:play, ok
  Movement:
    [0,E]:[1,1,1]->[1,2,2]
    [1,P]:[1,3,1]->[2,4,1]
    [2,P]:[1,5,1]->[2,5,2]
    [4,P]:[2,5,1]->[2,4,2]
    [6,P]:[3,4,1]->[2,3,2]
    [7,M]:[4,2,1]->[5,1,1]
    [10,B]:[5,2,1]->[5,1,3]
      reproduced [12,B] at [5,1,4]
      destroyed [7,M] at [5,1,1]
    [11,P]:[5,5,1]->[4,4,3]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   *E--   ----   ----   ----   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    P---   ----   WP--   PP--   -P--   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   ----   O---   -W--   M---   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   -*--   ----   P*P-   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    -WBB   ----   *---   P---   -W--   
->move(SE)
  state:5.0, mode:play, ok
  Movement:
    [0,E]:[1,2,2]->[2,3,3]
    [2,P]:[2,5,2]->[1,5,1]
    [3,P]:[5,4,1]->[1,4,1]
    [6,P]:[2,3,2]->[3,2,1]
    [8,P]:[2,1,1]->[1,5,2]
    [12,B]:[5,1,4]->[5,3,2]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   *---   ----   P---   PP--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    ----   ----   W-E-   PP--   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   P---   O---   -W--   M---   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   -*--   ----   P*P-   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    -WB-   ----   *B--   ----   -W--   
->move(SE)
  state:6.0, mode:play, ok
  Movement:
    [0,E]:[2,3,3]->[3,4,1]
    [1,P]:[2,4,1]->[1,3,1]
    [3,P]:[1,4,1]->[1,3,2]
    [5,M]:[3,5,1]->[2,1,1]
      reproduced [13,M] at [2,1,2]
    [6,P]:[3,2,1]->[2,1,3]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   *---   PP--   ----   PP--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    MMP-   ----   W---   -P--   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   ----   O---   EW--   ----   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   -*--   ----   P*P-   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    -WB-   ----   *B--   ----   -W--   
->play
  state:6.1, mode:play, error
  To start a new mission, please abort the current one first.
->move(N)
  state:7.0, mode:play, ok
  Explorer got lost in space - out of fuel at Sector:2:4
  The game has ended. You can start a new game.
  Movement:
    [0,E]:[3,4,1]->[2,4,1]
    [2,P]:[1,5,1]->[1,1,1]
    [4,P]:[2,4,2]->[2,5,1]
    [8,P]:[1,5,2]->[1,1,2]
    [10,B]:[5,1,3]->[5,1,1]
    [13,M]:[2,1,2]->[1,2,2]
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    PP--   *M--   PP--   ----   ----   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    M-P-   ----   W---   ----   P---   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   ----   O---   -W--   ----   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    W---   -*--   ----   P*P-   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   ----   *B--   ----   -W--   
->wormhole
  state:7.1, error
  Negative on that request:no mission in progress.
->move(SE)
  state:7.2, error
  Negative on that request:no mission in progress.
->move(N)
  state:7.3, error
  Negative on that request:no mission in progress.
->move(S)
  state:7.4, error
  Negative on that request:no mission in progress.
->move(NE)
  state:7.5, error
  Negative on that request:no mission in progress.
->move(S)
  state:7.6, error
  Negative on that request:no mission in progress.
->move(W)
  state:7.7, error
  Negative on that request:no mission in progress.
->move(SE)
  state:7.8, error
  Negative on that request:no mission in progress.
->move(W)
  state:7.9, error
  Negative on that request:no mission in progress.
->move(S)
  state:7.10, error
  Negative on that request:no mission in progress.
->wormhole
  state:7.11, error
  Negative on that request:no mission in progress.
->move(E)
  state:7.12, error
  Negative on that request:no mission in progress.
->test(11,20,20,38,50)
  state:8.0, mode:test, ok
  Movement:none
  Sectors:
    [1,1]->[0,E],[1,B],-,-
    [1,2]->[2,P],[3,P],[-6,W],-
    [1,3]->[4,J],[-7,Y],-,-
    [1,4]->[-10,*],-,-,-
    [1,5]->[5,B],[-8,Y],-,-
    [2,1]->[6,B],-,-,-
    [2,2]->[7,B],[8,P],[-4,Y],-
    [2,3]->[9,B],-,-,-
    [2,4]->-,-,-,-
    [2,5]->-,-,-,-
    [3,1]->[10,J],[-9,Y],-,-
    [3,2]->-,-,-,-
    [3,3]->[-1,O],-,-,-
    [3,4]->[11,B],[-5,Y],-,-
    [3,5]->[12,A],[13,J],-,-
    [4,1]->[14,B],-,-,-
    [4,2]->-,-,-,-
    [4,3]->[15,J],-,-,-
    [4,4]->[-11,W],-,-,-
    [4,5]->[16,P],-,-,-
    [5,1]->[17,B],[-2,W],-,-
    [5,2]->[18,P],[-3,*],-,-
    [5,3]->[19,A],-,-,-
    [5,4]->[20,J],-,-,-
    [5,5]->[21,B],-,-,-
  Descriptions:
    [-11,W]->
    [-10,*]->Luminosity:5
    [-9,Y]->Luminosity:2
    [-8,Y]->Luminosity:2
    [-7,Y]->Luminosity:2
    [-6,W]->
    [-5,Y]->Luminosity:2
    [-4,Y]->Luminosity:2
    [-3,*]->Luminosity:5
    [-2,W]->
    [-1,O]->
    [0,E]->fuel:3/3, life:3/3, landed?:F
    [1,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [2,P]->attached?:F, support_life?:F, visited?:F, turns_left:2
    [3,P]->attached?:F, support_life?:F, visited?:F, turns_left:2
    [4,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:1
    [5,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
    [6,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
    [7,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [8,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [9,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
    [10,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:0
    [11,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [12,A]->turns_left:1
    [13,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:1
    [14,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [15,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:1
    [16,P]->attached?:F, support_life?:F, visited?:F, turns_left:2
    [17,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
    [18,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [19,A]->turns_left:0
    [20,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:0
    [21,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
  Deaths This Turn:none
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    EB--   PPW-   JY--   *---   BY--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    B---   BPY-   B---   ----   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    JY--   ----   O---   BY--   AJ--   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    B---   ----   J---   W---   P---   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   P*--   A---   J---   B---   
->pass
  state:9.0, mode:test, ok
  Movement:
    [1,B]:[1,1,2]->[1,5,3]
    [7,B]:[2,2,1]->[2,3,2]
    [10,J]:[3,1,1]->[3,2,1]
    [11,B]:[3,4,1]->[4,4,2]
    [14,B]:[4,1,1]->[5,5,2]
    [19,A]:[5,3,1]->[5,4,2]
      destroyed [20,J] at [5,4,1]
  Sectors:
    [1,1]->[0,E],-,-,-
    [1,2]->[2,P],[3,P],[-6,W],-
    [1,3]->[4,J],[-7,Y],-,-
    [1,4]->[-10,*],-,-,-
    [1,5]->[5,B],[-8,Y],[1,B],-
    [2,1]->[6,B],-,-,-
    [2,2]->-,[8,P],[-4,Y],-
    [2,3]->[9,B],[7,B],-,-
    [2,4]->-,-,-,-
    [2,5]->-,-,-,-
    [3,1]->-,[-9,Y],-,-
    [3,2]->[10,J],-,-,-
    [3,3]->[-1,O],-,-,-
    [3,4]->-,[-5,Y],-,-
    [3,5]->[12,A],[13,J],-,-
    [4,1]->-,-,-,-
    [4,2]->-,-,-,-
    [4,3]->[15,J],-,-,-
    [4,4]->[-11,W],[11,B],-,-
    [4,5]->[16,P],-,-,-
    [5,1]->[17,B],[-2,W],-,-
    [5,2]->[18,P],[-3,*],-,-
    [5,3]->-,-,-,-
    [5,4]->-,[19,A],-,-
    [5,5]->[21,B],[14,B],-,-
  Descriptions:
    [-11,W]->
    [-10,*]->Luminosity:5
    [-9,Y]->Luminosity:2
    [-8,Y]->Luminosity:2
    [-7,Y]->Luminosity:2
    [-6,W]->
    [-5,Y]->Luminosity:2
    [-4,Y]->Luminosity:2
    [-3,*]->Luminosity:5
    [-2,W]->
    [-1,O]->
    [0,E]->fuel:3/3, life:3/3, landed?:F
    [1,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:1
    [2,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [3,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [4,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:0
    [5,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [6,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
    [7,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:0
    [8,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [9,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
    [10,J]->fuel:4/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:2
    [11,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:2
    [12,A]->turns_left:0
    [13,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:0
    [14,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:0
    [15,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:0
    [16,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [17,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
    [18,P]->attached?:T, support_life?:F, visited?:F, turns_left:N/A
    [19,A]->turns_left:1
    [21,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
  Deaths This Turn:
    [20,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:2/2, turns_left:N/A,
      Janitaur got destroyed by asteroid (id: 19) at Sector:5:4
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    E---   PPW-   JY--   *---   BYB-   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    B---   -PY-   BB--   ----   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   J---   O---   -Y--   AJ--   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    ----   ----   J---   WB--   P---   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   P*--   ----   -A--   BB--   
->move(E)
  state:10.0, mode:test, ok
  Movement:
    [0,E]:[1,1,1]->[1,2,4]
    [4,J]:[1,3,1]
    [5,B]:[1,5,1]->[5,5,3]
    [7,B]:[2,3,2]->[3,3,2]
    [12,A]:[3,5,1]->[4,5,2]
    [13,J]:[3,5,2]->[2,4,1]
    [14,B]:[5,5,2]->[1,4,2]
      reproduced [22,B] at [1,4,3]
    [15,J]:[4,3,1]->[5,4,1]
      destroyed [19,A] at [5,4,2]
  Sectors:
    [1,1]->-,-,-,-
    [1,2]->[2,P],[3,P],[-6,W],[0,E]
    [1,3]->[4,J],[-7,Y],-,-
    [1,4]->[-10,*],[14,B],[22,B],-
    [1,5]->-,[-8,Y],[1,B],-
    [2,1]->[6,B],-,-,-
    [2,2]->-,[8,P],[-4,Y],-
    [2,3]->[9,B],-,-,-
    [2,4]->[13,J],-,-,-
    [2,5]->-,-,-,-
    [3,1]->-,[-9,Y],-,-
    [3,2]->[10,J],-,-,-
    [3,3]->[-1,O],-,-,-
    [3,4]->-,[-5,Y],-,-
    [3,5]->-,-,-,-
    [4,1]->-,-,-,-
    [4,2]->-,-,-,-
    [4,3]->-,-,-,-
    [4,4]->[-11,W],[11,B],-,-
    [4,5]->[16,P],[12,A],-,-
    [5,1]->[17,B],[-2,W],-,-
    [5,2]->[18,P],[-3,*],-,-
    [5,3]->-,-,-,-
    [5,4]->[15,J],-,-,-
    [5,5]->[21,B],-,[5,B],-
  Descriptions:
    [-11,W]->
    [-10,*]->Luminosity:5
    [-9,Y]->Luminosity:2
    [-8,Y]->Luminosity:2
    [-7,Y]->Luminosity:2
    [-6,W]->
    [-5,Y]->Luminosity:2
    [-4,Y]->Luminosity:2
    [-3,*]->Luminosity:5
    [-2,W]->
    [-1,O]->
    [0,E]->fuel:2/3, life:3/3, landed?:F
    [1,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:0
    [2,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [3,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [4,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:2
    [5,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:1
    [6,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [8,P]->attached?:T, support_life?:T, visited?:F, turns_left:N/A
    [9,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [10,J]->fuel:4/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:1
    [11,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:1
    [12,A]->turns_left:0
    [13,J]->fuel:4/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:0
    [14,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
    [15,J]->fuel:4/5, load:1/2, actions_left_until_reproduction:1/2, turns_left:0
    [16,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [17,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [18,P]->attached?:T, support_life?:F, visited?:F, turns_left:N/A
    [21,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [22,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
  Deaths This Turn:
    [7,B]->fuel:1/3, actions_left_until_reproduction:0/1, turns_left:N/A,
      Benign got devoured by blackhole (id: -1) at Sector:3:3
    [19,A]->turns_left:N/A,
      Asteroid got imploded by janitaur (id: 15) at Sector:5:4
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   PPWE   JY--   *BB-   -YB-   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    B---   -PY-   B---   J---   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    -Y--   J---   O---   -Y--   ----   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    ----   ----   ----   WB--   PA--   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   P*--   ----   J---   B-B-   
->move(NE)
  state:11.0, mode:test, ok
  Movement:
    [0,E]:[1,2,4]->[5,3,1]
    [1,B]:[1,5,3]->[2,4,2]
      reproduced [23,B] at [2,4,3]
    [2,P]:[1,2,1]->[2,2,1]
    [3,P]:[1,2,2]->[2,1,2]
    [6,B]:[2,1,1]->[3,1,1]
    [9,B]:[2,3,1]->[3,4,1]
    [12,A]:[4,5,2]->[3,5,1]
    [13,J]:[2,4,1]->[3,4,3]
    [15,J]:[5,4,1]->[5,5,2]
    [16,P]:[4,5,1]->[5,5,4]
    [17,B]:[5,1,1]->[3,4,4]
    [21,B]:[5,5,1]->[5,1,1]
  Sectors:
    [1,1]->-,-,-,-
    [1,2]->-,-,[-6,W],-
    [1,3]->[4,J],[-7,Y],-,-
    [1,4]->[-10,*],[14,B],[22,B],-
    [1,5]->-,[-8,Y],-,-
    [2,1]->-,[3,P],-,-
    [2,2]->[2,P],[8,P],[-4,Y],-
    [2,3]->-,-,-,-
    [2,4]->-,[1,B],[23,B],-
    [2,5]->-,-,-,-
    [3,1]->[6,B],[-9,Y],-,-
    [3,2]->[10,J],-,-,-
    [3,3]->[-1,O],-,-,-
    [3,4]->[9,B],[-5,Y],[13,J],[17,B]
    [3,5]->[12,A],-,-,-
    [4,1]->-,-,-,-
    [4,2]->-,-,-,-
    [4,3]->-,-,-,-
    [4,4]->[-11,W],[11,B],-,-
    [4,5]->-,-,-,-
    [5,1]->[21,B],[-2,W],-,-
    [5,2]->[18,P],[-3,*],-,-
    [5,3]->[0,E],-,-,-
    [5,4]->-,-,-,-
    [5,5]->-,[15,J],[5,B],[16,P]
  Descriptions:
    [-11,W]->
    [-10,*]->Luminosity:5
    [-9,Y]->Luminosity:2
    [-8,Y]->Luminosity:2
    [-7,Y]->Luminosity:2
    [-6,W]->
    [-5,Y]->Luminosity:2
    [-4,Y]->Luminosity:2
    [-3,*]->Luminosity:5
    [-2,W]->
    [-1,O]->
    [0,E]->fuel:1/3, life:3/3, landed?:F
    [1,B]->fuel:2/3, actions_left_until_reproduction:1/1, turns_left:0
    [2,P]->attached?:T, support_life?:T, visited?:F, turns_left:N/A
    [3,P]->attached?:F, support_life?:F, visited?:F, turns_left:0
    [4,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:1
    [5,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:0
    [6,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:2
    [8,P]->attached?:T, support_life?:T, visited?:F, turns_left:N/A
    [9,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:2
    [10,J]->fuel:4/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:0
    [11,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:0
    [12,A]->turns_left:0
    [13,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:0/2, turns_left:1
    [14,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:1
    [15,J]->fuel:3/5, load:1/2, actions_left_until_reproduction:0/2, turns_left:2
    [16,P]->attached?:F, support_life?:F, visited?:F, turns_left:2
    [17,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:2
    [18,P]->attached?:T, support_life?:F, visited?:F, turns_left:N/A
    [21,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:0
    [22,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [23,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:2
  Deaths This Turn:none
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   --W-   JY--   *BB-   -Y--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    -P--   PPY-   ----   -BB-   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    BY--   J---   O---   BYJB   A---   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    ----   ----   ----   WB--   ----   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    BW--   P*--   E---   ----   -JBP   
->move(NW)
  state:12.0, mode:test, ok
  Explorer got lost in space - out of fuel at Sector:4:2
  The game has ended. You can start a new game.
  Movement:
    [0,E]:[5,3,1]->[4,2,1]
    [1,B]:[2,4,2]->[3,5,2]
    [3,P]:[2,1,2]->[1,2,1]
    [5,B]:[5,5,3]->[4,4,3]
      reproduced [24,B] at [4,4,4]
    [10,J]:[3,2,1]->[4,3,1]
    [11,B]:[4,4,2]->[4,5,1]
      reproduced [25,B] at [4,5,2]
    [12,A]:[3,5,1]->[2,4,1]
      destroyed [23,B] at [2,4,3]
    [21,B]:[5,1,1]->[3,5,1]
      reproduced [26,B] at [3,5,3]
    [22,B]:[1,4,3]->[2,4,2]
  Sectors:
    [1,1]->-,-,-,-
    [1,2]->[3,P],-,[-6,W],-
    [1,3]->[4,J],[-7,Y],-,-
    [1,4]->[-10,*],[14,B],-,-
    [1,5]->-,[-8,Y],-,-
    [2,1]->-,-,-,-
    [2,2]->[2,P],[8,P],[-4,Y],-
    [2,3]->-,-,-,-
    [2,4]->[12,A],[22,B],-,-
    [2,5]->-,-,-,-
    [3,1]->[6,B],[-9,Y],-,-
    [3,2]->-,-,-,-
    [3,3]->[-1,O],-,-,-
    [3,4]->[9,B],[-5,Y],[13,J],[17,B]
    [3,5]->[21,B],[1,B],[26,B],-
    [4,1]->-,-,-,-
    [4,2]->-,-,-,-
    [4,3]->[10,J],-,-,-
    [4,4]->[-11,W],-,[5,B],[24,B]
    [4,5]->[11,B],[25,B],-,-
    [5,1]->-,[-2,W],-,-
    [5,2]->[18,P],[-3,*],-,-
    [5,3]->-,-,-,-
    [5,4]->-,-,-,-
    [5,5]->-,[15,J],-,[16,P]
  Descriptions:
    [-11,W]->
    [-10,*]->Luminosity:5
    [-9,Y]->Luminosity:2
    [-8,Y]->Luminosity:2
    [-7,Y]->Luminosity:2
    [-6,W]->
    [-5,Y]->Luminosity:2
    [-4,Y]->Luminosity:2
    [-3,*]->Luminosity:5
    [-2,W]->
    [-1,O]->
    [1,B]->fuel:1/3, actions_left_until_reproduction:0/1, turns_left:2
    [2,P]->attached?:T, support_life?:T, visited?:F, turns_left:N/A
    [3,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [4,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:1/2, turns_left:0
    [5,B]->fuel:1/3, actions_left_until_reproduction:1/1, turns_left:0
    [6,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:1
    [8,P]->attached?:T, support_life?:T, visited?:F, turns_left:N/A
    [9,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:1
    [10,J]->fuel:3/5, load:0/2, actions_left_until_reproduction:0/2, turns_left:1
    [11,B]->fuel:2/3, actions_left_until_reproduction:1/1, turns_left:1
    [12,A]->turns_left:2
    [13,J]->fuel:5/5, load:0/2, actions_left_until_reproduction:0/2, turns_left:0
    [14,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [15,J]->fuel:3/5, load:1/2, actions_left_until_reproduction:0/2, turns_left:1
    [16,P]->attached?:F, support_life?:F, visited?:F, turns_left:1
    [17,B]->fuel:3/3, actions_left_until_reproduction:0/1, turns_left:1
    [18,P]->attached?:T, support_life?:F, visited?:F, turns_left:N/A
    [21,B]->fuel:2/3, actions_left_until_reproduction:1/1, turns_left:0
    [22,B]->fuel:2/3, actions_left_until_reproduction:0/1, turns_left:1
    [24,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [25,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
    [26,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:0
  Deaths This Turn:
    [0,E]->fuel:0/3, life:0/3, landed?:F,
      Explorer got lost in space - out of fuel at Sector:4:2
    [23,B]->fuel:3/3, actions_left_until_reproduction:1/1, turns_left:N/A,
      Benign got destroyed by asteroid (id: 12) at Sector:2:4
    (1:1)  (1:2)  (1:3)  (1:4)  (1:5)  
    ----   P-W-   JY--   *B--   -Y--   
    (2:1)  (2:2)  (2:3)  (2:4)  (2:5)  
    ----   PPY-   ----   AB--   ----   
    (3:1)  (3:2)  (3:3)  (3:4)  (3:5)  
    BY--   ----   O---   BYJB   BBB-   
    (4:1)  (4:2)  (4:3)  (4:4)  (4:5)  
    ----   ----   J---   W-BB   BB--   
    (5:1)  (5:2)  (5:3)  (5:4)  (5:5)  
    -W--   P*--   ----   ----   -J-P   
  Explorer got lost in space - out of fuel at Sector:4:2
  The game has ended. You can start a new game.
->move(NW)
  state:12.1, error
  Negative on that request:no mission in progress.
->wormhole
  state:12.2, error
  Negative on that request:no mission in progress.
->wormhole
  state:12.3, error
  Negative on that request:no mission in progress.
->move(N)
  state:12.4, error
  Negative on that request:no mission in progress.
->abort
  state:12.5, error
  Negative on that request:no mission in progress.
->move(W)
  state:12.6, error
  Negative on that request:no mission in progress.
->move(NW)
  state:12.7, error
  Negative on that request:no mission in progress.
->move(NW)
  state:12.8, error
  Negative on that request:no mission in progress.
->wormhole
  state:12.9, error
  Negative on that request:no mission in progress.
->move(N)
  state:12.10, error
  Negative on that request:no mission in progress.
->wormhole
  state:12.11, error
  Negative on that request:no mission in progress.
->move(NW)
  state:12.12, error
  Negative on that request:no mission in progress.
->move(N)
  state:12.13, error
  Negative on that request:no mission in progress.
->liftoff
  state:12.14, error
  Negative on that request:no mission in progress.
->liftoff
  state:12.15, error
  Negative on that request:no mission in progress.
->pass
  state:12.16, error
  Negative on that request:no mission in progress.
->abort
  state:12.17, error
  Negative on that request:no mission in progress.
->move(SE)
  state:12.18, error
  Negative on that request:no mission in progress.
->move(SE)
  state:12.19, error
  Negative on that request:no mission in progress.
->move(NW)
  state:12.20, error
  Negative on that request:no mission in progress.
->move(SW)
  state:12.21, error
  Negative on that request:no mission in progress.
->abort
  state:12.22, error
  Negative on that request:no mission in progress.
->move(E)
  state:12.23, error
  Negative on that request:no mission in progress.
